mkosi/backend.py

```diff
@@ -587,9 +589,16 @@ def write_grub_config(args: CommandLineArguments, root: Path) -> None:
     else:
 
         def jj(line: str) -> str:
-            if line.startswith("GRUB_CMDLINE_LINUX="):
+            if line.startswith(("GRUB_CMDLINE_LINUX=", "#GRUB_CMDLINE_LINUX=")):  # GENTOO:
                 return grub_cmdline
             if args.qemu_headless:
+                # GENTOO: uses the catch-all version GRUB_TERMINAL
+                # if we check here we end up with this *dangling* comment in
+                # the config file: "Uncomment to disable graphical terminal (grub-pc only)"
+                # if "GRUB_TERMINAL" in line:
+                #     return 'GRUB_TERMINAL="console serial"'
+                # see alternative approach below
+
```

Comment:
I think I'd like to keep this all in one place. Another possibility would be a regex matching on something like `GRUB_TERMINAL(_INPUT|_OUTPUT|)`